packages/lisk-transactions/test/transactions/base.ts

```diff
@@ -17,647 +17,498 @@ import { BaseTransaction } from '../../src/transactions/base';
+			expect(baseTransaction)
 				.to.have.property('id')
 				.and.be.a('string');
 		});
 
-		it('should have recipientId string', () => {
-			return expect(baseTransaction)
+		it('should have recipientId string', async () => {
+			expect(baseTransaction)
 				.to.have.property('recipientId')
 				.and.be.a('string');
 		});
 
-		it('should have recipientPublicKey string', () => {
-			return expect(baseTransaction)
+		it('should have recipientPublicKey string', async () => {
+			expect(baseTransaction)
 				.to.have.property('recipientPublicKey')
 				.and.be.a('string');
 		});
 
-		it('should have senderId string', () => {
-			return expect(baseTransaction)
+		it('should have senderId string', async () => {
+			expect(baseTransaction)
 				.to.have.property('senderId')
 				.and.be.a('string');
 		});
 
-		it('should have senderPublicKey string', () => {
-			return expect(baseTransaction)
+		it('should have senderPublicKey string', async () => {
+			expect(baseTransaction)
 				.to.have.property('senderPublicKey')
 				.and.be.a('string');
 		});
 
-		it('should have signature string', () => {
-			return expect(baseTransaction)
+		it('should have signature string', async () => {
+			expect(baseTransaction)
 				.to.have.property('senderPublicKey')
 				.and.be.a('string');
 		});
 
-		it('should have signSignature string', () => {
-			return expect(baseTransaction)
+		it('should have signSignature string', async () => {
+			expect(baseTransaction)
 				.to.have.property('senderPublicKey')
 				.and.be.a('string');
 		});
 
-		it('should have signatures array', () => {
-			return expect(baseTransaction)
+		it('should have signatures array', async () => {
+			expect(baseTransaction)
 				.to.have.property('signatures')
 				.and.be.a('array');
 		});
 
-		it('should have timestamp number', () => {
-			return expect(baseTransaction)
+		it('should have timestamp number', async () => {
+			expect(baseTransaction)
 				.to.have.property('timestamp')
 				.and.be.a('number');
 		});
 
-		it('should have type number', () => {
-			return expect(baseTransaction)
+		it('should have type number', async () => {
+			expect(baseTransaction)
 				.to.have.property('type')
 				.and.be.a('number');
 		});
 
-		it('should have receivedAt Date', () => {
-			return expect(baseTransaction)
+		it('should have receivedAt Date', async () => {
+			expect(baseTransaction)
 				.to.have.property('type')
 				.and.be.a('number');
 		});
 
-		it('should have isMultisignature boolean', () => {
-			return expect(baseTransaction)
+		it('should have isMultisignature boolean', async () => {
+			expect(baseTransaction)
 				.to.have.property('isMultisignature')
 				.and.be.a('boolean');
 		});
 
-		describe('when transaction has empty `signatures`', () => {
-			it('should set isMultisignature to false', () => {
-				return expect(baseTransaction.isMultisignature).to.be.false;
-			});
+		it('should set isMultisignature to true with a multisignature transaction', async () => {
```

Comment:
Do we set isMultisignature to true even when it doesn't have signatures but the account is a multisignature?

Reply:
not in `constructor` as we don't have access to account state. however it is a good point as `isMultisignature` will be checked in `isExpired` to increase time until expiry. currently `apply` `undo` and `verifyAgainstTransactions` have access to account state. thoughts? @SargeKhan @shuse2